### ISSUE TITLE
[Inquiry]:Integrate this script with datasone SMTC

### ISSUE THREAD

**Issue by NaiveInvestigator** [qwerty12/mpv-taskbar-buttons#1]
lol yes ik that was fast just happened to click and saw your script the moment you added in the wiki. Love it already!!

I was curious that if this could use the exe that comes with [datasone's script](https://github.com/datasone/MPVMediaControl) instead of the dll in the future. or maybe modify the exe so that it can do just that.

Edit: Dunno just a random suggestion and has to do nothing with the fear ppl have of dll lol really.

**Comment by qwerty12** [comment]
Thanks!

Unfortunately (well, it makes sense usually; this is a special case), Windows will only notify the process that the window belongs to that one of the taskbar buttons have been clicked. The only way to get notified of these messages from an external program is to use `SetWindowsHookEx` from said program which involves, wait for it, Windows loading a custom DLL into the target process.

The SMTC programs are different in that they're responding to events put out by mpv and the Windows integration there isn't tied to the actual mpv entry in the taskbar, if I'm talking any sense. As far as I can tell, to do the same in an external process without any DLLs, you would have to make a new window separate from mpv's and attach controls to that, which would be pointless.

The pure-luajit branch of this repo has a version of this that is written in nothing but Lua (LuaJIT-infused Lua, rather) and needs no external DLL but I had to abandon it because:

* it can't keep up if there's a high amount of messages to process and mpv ends up crashing - I could trigger a crash by moving my mouse very quickly

* hook callbacks should be fast and quick to return or mpv GUI processing will be slowed down. The problem is that the LuaJIT manual outright says that its callbacks are slow and not to use them for performance-sensitive work...

* edit: the callback also runs in a different thread to the one the script runs on, so referencing existing things in the Lua code just meant the script was bound to crash anyway

Even if I managed to fix the crash, I couldn't make it any faster. So I was going to have to include a DLL of some sort: this could be one with everything compiled, and I include the source separately, but changing how that part works as an end-user would require setting up Visual Studio etc., or I could provide a DLL that compiles C code on demand and allows the user to change how the hook functions with just Notepad
